java/mykola-melnyk/src/com/zoolatech/lecture4/tasks/_4/WeightUnits.java

```diff
@@ -0,0 +1,90 @@
+package com.zoolatech.lecture4.tasks._4;
+
+public enum WeightUnits {
+    GRAM("g"), KILOGRAM("kg"), OUNCE("oz"),
+    POUND("lb"), TON("t");
+    private final String abbreviation;
+    WeightUnits(String abbreviation) {
+        this.abbreviation = abbreviation;
+    }
+
+    public String getAbbreviation() {
+        return abbreviation;
+    }
+
+    public double convertFrom(double value, WeightUnits sourceUnit) {
+        double result = 0;
+        switch (this) {
+            case GRAM -> {
+                result = switch (sourceUnit) {
```

Comment:
you can return the value directly from the switch expression, without storing it to a local variable

Reply:
Done